Make props file work for builds using any prefix/DLL combinations

vadz opened the pull request:
The existing wxwidgets.props wasn't particularly useful as it couldn't
be used with with wx DLLs when the user project configuration were not
called "DLL Debug" or "DLL Release" (and it makes little sense for the
user application to use "DLL" in its configuration names just because it
happens to use wx as DLL, of course).

It also couldn't be used with the libraries built using nmake with a
custom COMPILER_PREFIX, which, significantly, includes the official MSW
binaries.

Try to fix both problems by checking for the actually existing libraries
and using whichever ones we find. This is somewhat surprising for a
project file to do, but it seems like it should result in the most
useful behaviour in practice as it allows user projects importing this
file to work out of the box in all of the following situations:

  - Using official wxMSW binaries.
  - Building wxMSW from source without any customization.
  - Building wxMSW from source using reasonable custom compiler prefix.

---

I'm rather ashamed to admit this, but it turns out that the advice for building wx applications with MSVC I was giving since quite some time didn't actually work with our own official binaries and not just for one, but for 2 reasons:

1. It didn't use the DLLs unless the application project configuration happened to be called "DLL Debug/Release" which is, of course, not the case by default (and shouldn't be required).
1. It didn't look for the libraries under the right directory anyhow, as it always used `lib\vc_lib` instead of `lib\vc14x_dll`.

I started fixing this by creating a `.props` file for the official DLLs, but then realized that we could actually do everything from the same file, using MSBuild `Exists()` condition. I don't know if I didn't create a monster here, but at least this seems to work fine in all situations (and I've tested it with a few). So what do you think of doing it like this?

vadz committed:
Make props file work for builds using any prefix/DLL combinations

The existing wxwidgets.props wasn't particularly useful as it couldn't
be used with with wx DLLs when the user project configuration were not
called "DLL Debug" or "DLL Release" (and it makes little sense for the
user application to use "DLL" in its configuration names just because it
happens to use wx as DLL, of course).

It also couldn't be used with the libraries built using nmake with a
custom COMPILER_PREFIX, which, significantly, includes the official MSW
binaries.

Try to fix both problems by checking for the actually existing libraries
and using whichever ones we find. This is somewhat surprising for a
project file to do, but it seems like it should result in the most
useful behaviour in practice as it allows user projects importing this
file to work out of the box in all of the following situations:

  - Using official wxMSW binaries.
  - Building wxMSW from source without any customization.
  - Building wxMSW from source using reasonable custom compiler prefix.

doscott commented:
I pulled this into a VS2019 project and during a rebuild it picked up a macro redefinition of wxMSVC_VERSION_AUTO, which I had defined, and picked up the proper include and lib directories. Removing my definition and my settings of include and library paths for wx resulted with a successful build. This will definitely simplify setting up projects.

vadz committed:
Explain in more details how to use wxWidgets.props

Also describe a manual change which needs to be done at the project
level when creating a new empty project in MSVS.
